src/psbtv2.ts

```diff
@@ -0,0 +1,910 @@
+        const key = bw.render().toString('hex');
+        bw.writeBytes(bip32.masterFingerprint);
+        bw.writeBytes(parseDerivationPathNodesToBytes(bip32.path))
+        map.set(key, bw.render());
+      }
+    }
+
+    this.inputMaps.push(map)
+  }
+
+
+  // Removes an input-map from inputMaps
+  public deleteInput (index:number) {
+    // TODO: Maybe this needs to check PSBT_GLOBAL_TX_MODIFIABLE before
+    // performing the operation.
+    const newInputs = this.inputMaps.filter((_, i) => i !== index);
+    this.PSBT_GLOBAL_INPUT_COUNT = newInputs.length;
+    this.inputMaps = newInputs;
+  }
+
+
+  public addOutput ({
+    amount,
+    script,
+    redeemScript,
+    witnessScript,
+    bip32Derivation,
+  }: {
+    amount:number;
+    script:Buffer;
+    redeemScript?:Buffer,
+    witnessScript?:Buffer,
+    bip32Derivation?:{
+      pubkey:Buffer;
+      masterFingerprint:Buffer;
+      path:string;
+    }[],
+  }) {
+    // TODO: Maybe this needs to check PSBT_GLOBAL_TX_MODIFIABLE before
+    // performing the operation.
+    const map = new Map<Key, Value>();
+    const bw = new BufferWriter();
+    bw.writeI64(amount);
+    map.set(KeyType.PSBT_OUT_AMOUNT, bw.render());
+    bw.writeBytes(script);
+    map.set(KeyType.PSBT_OUT_SCRIPT, bw.render());
+
+    if (redeemScript) {
+      bw.writeBytes(script);
+      map.set(KeyType.PSBT_OUT_REDEEM_SCRIPT, bw.render());
+    }
+
+    if (witnessScript) {
+      bw.writeBytes(script);
+      map.set(KeyType.PSBT_OUT_WITNESS_SCRIPT, bw.render());
+    }
+
+    if (bip32Derivation) {
+      for (const bip32 of bip32Derivation) {
+        bw.writeString(KeyType.PSBT_OUT_BIP32_DERIVATION, 'hex');
+        bw.writeBytes(bip32.pubkey)
+        const key = bw.render().toString('hex');
+        bw.writeBytes(bip32.masterFingerprint);
+        bw.writeBytes(parseDerivationPathNodesToBytes(bip32.path))
+        map.set(key, bw.render());
+      }
+    }
+
+    this.outputMaps.push(map);
+  }
+
+
+  // Removes an output-map from outputMaps
+  public deleteOutput (index:number) {
+    // TODO: Maybe this needs to check PSBT_GLOBAL_TX_MODIFIABLE before
+    // performing the operation.
+    const newOutputs = this.outputMaps.filter((_, i) => i !== index);
+    this.PSBT_GLOBAL_OUTPUT_COUNT = newOutputs.length;
+    this.outputMaps = newOutputs;
+  }
+
+
+  // Return the current state of the psbt as a string in the specified format.
+  public serialize (format:'base64'|'hex' = 'base64') {
+    // Build hex string from maps
+    let bw = new BufferWriter();
+    bw.writeBytes(PSBT_MAGIC_BYTES);
+    serializeMap(this.globalMap, bw);
+    
+    for (const map of this.inputMaps) {
+      serializeMap(map, bw);
+    }
+
+    for (const map of this.outputMaps) {
+      serializeMap(map, bw);
+    }
+
+    return bw.render().toString(format);
+  }
+
+
+  // Attempt to return a PsbtV2 by converting from a PsbtV0 string or Buffer
+  static FromV0 (psbt:string|Buffer):PsbtV2 {
```

Comment:
I guess this is capitalized because it's `static`? Other static methods in the library are still camel cased. I've not seen pascal casing for anything but objects/classes personally.